src/java/org/apache/cassandra/schema/Schema.java

```diff
@@ -224,6 +225,15 @@ public ColumnFamilyStore getColumnFamilyStoreInstance(TableId id)
                : null;
     }
 
+    public ColumnFamilyStore getColumnFamilyStoreInstance(TableMetadata metadata)
```

Comment:
The behaviour of this is different from id lookup. Could we give it a different name?

Reply:
actually, after applying your fixes this method is not needed any longer; so I will just add some comment to the original method